blockdag/timesource.go

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2013-2014 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package blockdag
+
+import (
+	"time"
+)
+
+// TimeSource is the interface to access time.
+type TimeSource interface {
+	// Now returns the current time.
+	Now() time.Time
+}
+
+// timeSource provides an implementation of the TimeSource interface
+// that simply returns the current local time.
+type timeSource struct{}
+
+// Ensure the timeSource type implements the TimeSource interface.
```

Comment:
This looks somewhat redundant.

I mean, it might be an ok practice to do it everywhere, but we are not doing it anywhere else, so why bother here?